### DISCUSSION TITLE
Program fails to start as MariaDB has no password set so the container fails to start

### DISCUSSION THREAD

**Discussion by ethernet6** [serversideup/financial-freedom#110] [Q&A]
After setting the env with MARIADB_ROOT_PASSWORD etc. parameters this still does not address the error I faced when I started the docker compose file. The images are created for the program via `Docker compose up` but the MARIADB container's password is not set at the configured entry point. As a person new to Docker I am unable to fix this and or make the database run to support the greater application, it therefore fails to load on localhost:3000 and MariaDB exits with code 1.

Anyone encountered this error?

**Comment by danpastori** [comment]
@jaydrogers might have a little more insight on this. We will take a look!

**Comment by RokSiEu** [comment]
I can confirm, I'm using docker and portainer to manage my docker, and today when I've followed https://serversideup.net/open-source/financial-freedom/docs/community/contributing#running-locally-in-your-development it failed

> 2024-01-20 19:39:50+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.11.6+maria~ubu2204 started.
2024-01-20 19:39:51+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-01-20 19:39:51+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.11.6+maria~ubu2204 started.
2024-01-20 19:39:51+00:00 [ERROR] [Entrypoint]: Database is uninitialized and password option is not specified
	You need to specify one of MARIADB_ROOT_PASSWORD, MARIADB_ROOT_PASSWORD_HASH, MARIADB_ALLOW_EMPTY_ROOT_PASSWORD and MARIADB_RANDOM_ROOT_PASSWORD

**Comment by jaydrogers** [comment]
**Just an update:** We're thinking of converting everything to SQLite to solve this problem. Then people don't need the complexities of managing a database

**Comment by M3teora** [comment]
> After setting the env with MARIADB_ROOT_PASSWORD etc. parameters this still does not address the error I faced when I started the docker compose file. The images are created for the program via `Docker compose up` but the MARIADB container's password is not set at the configured entry point. As a person new to Docker I am unable to fix this and or make the database run to support the greater application, it therefore fails to load on localhost:3000 and MariaDB exits with code 1.
> 
> Anyone encountered this error?

Hi, I've added this line, and it worked 
  mariadb:
    image: mariadb:10.11
    volumes:
      - database:/var/lib/mysql
    **environment:
      - MYSQL_ROOT_PASSWORD=some_password**
